### PR TITLE
fixed keyboardDismissBehavior on scroll without a drag

### LINKED ISSUE #154515: Autocomplete field options in the SingleChildScrollView remains open even when field goes out of view.
### Steps to reproduce

1. Wrap Autocomplete field in the SingleChildScrollView.
2. Click on the field as if you are selecting options.
3. Scroll to make field out of view. Remember to scroll by two fingers (not by clicking and dragging the scrollbar).
4. You will notice the options overlay remains open and follow the field which is not even visible.

### Expected results

Options should close as soon as field goes out of view.

### Actual results

It remains open.

### Code sample

https://dartpad.dev/?id=61735dfd0bbd1de1211510d4cdab387d

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

image

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Dart SDK 3.5.1 and Flutter SDK 3.24.1
```

</details>

### PR DESCRIPTION
**Pull request by nick9822**
fixes #154515, #150048 and other similar issues where user non-draggable scrolls (mouse wheel, two-fingers) should behave same as draggable ones.

In this PR, scrollUpdateNotification.dragDetails check is removed and it has a supporting test which simulates a scroll which does not produce a drag.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by goderbauer** [comment]
(triage): Could you please take a look at the analyzer check and fix what it is complaining about? Thank you!

**Comment by nick9822** [comment]
@goderbauer fixed it, now `Linux analyze` is passing. But `Windows build_tests_6_7` is failing but doesn't seem to be related to the newline commit.

File: packages/flutter/lib/src/widgets/single_child_scroll_view.dart [hunk 1]
```diff
@@ -276,7 +276,7 @@ class SingleChildScrollView extends StatelessWidget {
         child: scrollable,
         onNotification: (ScrollUpdateNotification notification) {
           final FocusScopeNode currentScope = FocusScope.of(context);
-          if (notification.dragDetails != null && !currentScope.hasPrimaryFocus && currentScope.hasFocus) {
+          if (!currentScope.hasPrimaryFocus && currentScope.hasFocus) {
```
Comment (victorsanni) [inline packages/flutter/lib/src/widgets/single_child_scroll_view.dart:279]: My understanding of what's happening here is that the widget is now always unfocused when it is being scrolled? If that's correct, might there be cases where that behavior is undesired?

Comment (nick9822) [reply packages/flutter/lib/src/widgets/single_child_scroll_view.dart:279]: @victorsanni Not that I am aware of, now it is as good as scroll with a drag which removes focus from the field when scrollbar is dragged. I found another issue #150048 which is similar to this and there is a discussion on this.

**Review by victorsanni — APPROVED** [review]
This PR is identical to https://github.com/flutter/flutter/pull/150317. While that PR was created earlier, the author of that PR has not responded in a while, and we'd like to go forward with this change (as per the conversation in the other PR).

cc @justinmc

File: packages/flutter/test/widgets/single_child_scroll_view_test.dart [hunk 2]
```diff
@@ -1118,4 +1120,47 @@ void main() {
 
     expect(tester.testTextInput.isVisible, isFalse);
   });
+
+testWidgets('keyboardDismissBehavior on scroll without a drag test', (WidgetTester tester) async {
```
Comment (victorsanni) [inline packages/flutter/test/widgets/single_child_scroll_view_test.dart:1124]:
```suggestion
  testWidgets('keyboardDismissBehavior on scroll without a drag test', (WidgetTester tester) async {
```

File: packages/flutter/test/widgets/single_child_scroll_view_test.dart [hunk 3]
```diff
@@ -1118,4 +1120,47 @@ void main() {
...
+              children: <Widget>[
+                Autocomplete<String>(
+                  optionsBuilder: (TextEditingValue textEditingValue) {
+                    return<String>['aardvark', 'bobcat', 'chameleon']
+                        .where((String option) {
+                      return option.contains(textEditingValue.text.toLowerCase());
+                    });
+                  },
+                ),
+                const SizedBox(height: 2000)
```
Comment (victorsanni) [inline packages/flutter/test/widgets/single_child_scroll_view_test.dart:1141]:
```suggestion
                const SizedBox(height: 2000),
```

File: packages/flutter/test/widgets/single_child_scroll_view_test.dart [hunk 4]
```diff
@@ -1118,4 +1120,47 @@ void main() {
...
+    await tester.enterText(find.byType(RawAutocomplete<String>),'aard');
+    await tester.pump();
+
+    expect(find.text('aardvark'), findsOneWidget);
+
+    final TestPointer testPointer = TestPointer(1, PointerDeviceKind.mouse);
+    final Offset scrollStart = tester.getCenter(find.byType(SingleChildScrollView));
+
+    testPointer.hover(scrollStart);
+    await tester.sendEventToBinding(testPointer.scroll(Offset(scrollStart.dx, scrollStart.dy-100)));
```
Comment (victorsanni) [inline packages/flutter/test/widgets/single_child_scroll_view_test.dart:1161]:
```suggestion
    await tester.sendEventToBinding(testPointer.scroll(Offset(scrollStart.dx, scrollStart.dy - 100)));
```

**Review by victorsanni — CHANGES_REQUESTED** [review]
LGTM pending formatting nits.

**Comment by nick9822** [comment]
@victorsanni did the formatting changes as suggested.

**Review by victorsanni — APPROVED** [review]

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Comment by priyankabambharoliya-official** [comment]
After this update, the issue occurs when we use keyboard dismiss behaviour on drag
`keyboardDismissBehavior: ScrollViewKeyboardDismissBehavior.onDrag`

see this video

https://github.com/user-attachments/assets/583417ea-4def-4da0-bad2-cf2bf59b4b9c

**Pull request merged**